core/src/main/java/org/bitcoinj/evolution/SimplifiedMasternodeListManager.java

```diff
@@ -507,83 +514,39 @@ public void onFirstSaveComplete() {
-        Preconditions.checkState(getMasternodeListCache().size() == 1);
-        Preconditions.checkState(getQuorumListCache(params.getLlmqChainLocks()).size() == 1);
-
-        new Thread(new Runnable() {
-            @Override
-            public void run() {
-
-                log.info("loading mnlistdiff bootstrap file: " + bootStrapFilePath != null ? bootStrapFilePath : "input stream");
-                Context.propagate(context);
-                //load the file
-                InputStream stream = bootStrapStream;
-                try {
-                    if(stream != null)
-                        stream.reset();
-                    stream = stream != null ? stream : new FileInputStream(bootStrapFilePath);
-                    byte[] buffer = new byte[(int) stream.available()];
-                    stream.read(buffer);
-
-                    isLoadingBootStrap = true;
-                    switch (bootStrapFileFormat) {
-                        case LLMQ_FORMAT_VERSION:
-                            SimplifiedMasternodeListDiff mnlistdiff = new SimplifiedMasternodeListDiff(params, buffer);
-                            processMasternodeListDiff(null, mnlistdiff, true);
-                            break;
-                        case QUORUM_ROTATION_FORMAT_VERSION:
-                            QuorumRotationInfo qrinfo = new QuorumRotationInfo(params, buffer);
-                            processQuorumRotationInfo(null, qrinfo, true);
-                            break;
-                        default:
-                            throw new IllegalArgumentException("file format " + bootStrapFileFormat + " is not supported");
-                    }
-                    bootStrapLoaded.set(true);
-                    log.info("finished loading mnlist bootstrap file");
-                } catch (VerificationException | IOException | IllegalStateException | NullPointerException x) {
-                    bootStrapLoaded.setException(x);
-                    log.info("failed loading mnlist bootstrap file" + x.getMessage());
-                } finally {
-                    isLoadingBootStrap = false;
-                    try {
-                        if (stream != null)
-                            stream.close();
-                        quorumState.requestAfterMNListReset();
-                        quorumRotationState.requestAfterMNListReset();
-                    } catch (IOException x) {
-
-                    } catch (BlockStoreException x) {
-                        throw new RuntimeException(x);
-                    }
-                }
-            }
-        }).start();
+    @Deprecated
+    public static void setBootStrapStream(InputStream mnlistdiffStream, InputStream qrinfoStream, int format) {
+        SimplifiedMasternodeListManager.mnlistdiffBootStrapStream = mnlistdiffStream;
+        SimplifiedMasternodeListManager.qrinfoBootStrapStream = qrinfoStream;
+        bootStrapFileFormat = format;
     }
 
     /**
```

Comment:
This file is refactored nicely, I think it's this one that was over 1k lines?

Reply:
Yes, this was 957 lines before we started Core 18 and it grew considerably by the first time you pointed out the size.